src/measure_theory/decomposition/radon_nikodym.lean

```diff
@@ -65,6 +68,168 @@ theorem absolutely_continuous_iff_with_density_radon_nikodym_deriv_eq
+    rw [← with_density_apply _ measurable_set.univ,
+        with_density_radon_nikodym_deriv_eq μ ν h],
+    exact measure_lt_top _ _ },
+end
+
 end measure
 
+namespace signed_measure
+
+include m
+
+open measure vector_measure
+
+/-- The Radon-Nikodym derivative between a signed measure and a positive measure.
+
+`radon_nikodym_deriv s μ` satisfies `μ.with_densityᵥ (s.radon_nikodym_deriv μ) = s`
+if and only if `s` is absolutely continuous with respect to `μ` and this fact is known as
+`measure_theory.signed_measure.absolutely_continuous_iff_with_density_radon_nikodym_deriv_eq`. -/
+def radon_nikodym_deriv (s : signed_measure α) (μ : measure α) : α → ℝ := λ x,
+(s.to_jordan_decomposition.pos_part.radon_nikodym_deriv μ x).to_real -
+(s.to_jordan_decomposition.neg_part.radon_nikodym_deriv μ x).to_real
+
+@[measurability]
+lemma measurable_radon_nikodym_deriv (s : signed_measure α) (μ : measure α) :
+  measurable (radon_nikodym_deriv s μ) :=
+begin
+  rw [radon_nikodym_deriv],
+  measurability,
+end
+
+lemma integrable_radon_nikodym_deriv (s : signed_measure α) (μ : measure α) :
+  integrable (radon_nikodym_deriv s μ) μ :=
+begin
+  refine integrable.sub _ _;
+  { split, measurability,
+    exact has_finite_integral_to_real_of_lintegral_ne_top
+      (lintegral_radon_nikodym_deriv_lt_top _ μ).ne }
+end
+
+theorem with_densityᵥ_radon_nikodym_deriv_eq
+  (s : signed_measure α) (μ : measure α) [sigma_finite μ]
+  (h : s ≪ μ.to_ennreal_vector_measure) :
+  μ.with_densityᵥ (s.radon_nikodym_deriv μ) = s :=
+begin
+  rw [absolutely_continuous_ennreal_iff,
+      (_ : μ.to_ennreal_vector_measure.ennreal_to_measure = μ),
+      total_variation_absolutely_continuous_iff] at h,
+  { ext1 i hi,
+    rw [with_densityᵥ_apply (integrable_radon_nikodym_deriv _ _) hi,
+        radon_nikodym_deriv, integral_sub,
+        with_density_radon_nikodym_deriv_to_real_eq h.1 hi,
+        with_density_radon_nikodym_deriv_to_real_eq h.2 hi],
+    { conv_rhs { rw ← s.to_signed_measure_to_jordan_decomposition },
+      erw vector_measure.sub_apply,
+      rw [to_signed_measure_apply_measurable hi, to_signed_measure_apply_measurable hi] },
+    all_goals { rw ← integrable_on_univ,
+      refine integrable_on.restrict _ measurable_set.univ,
+      refine ⟨_, has_finite_integral_to_real_of_lintegral_ne_top _⟩,
+      { measurability },
+      { rw set_lintegral_univ,
+        exact (lintegral_radon_nikodym_deriv_lt_top _ _).ne } } },
+  { exact equiv_measure.right_inv μ }
+end
+
+/-- The Radon-Nikodym theorem for signed measures. -/
+theorem absolutely_continuous_iff_with_densityᵥ_radon_nikodym_deriv_eq
+  (s : signed_measure α) (μ : measure α) [sigma_finite μ] :
+  s ≪ μ.to_ennreal_vector_measure ↔
+  μ.with_densityᵥ (s.radon_nikodym_deriv μ) = s :=
+⟨with_densityᵥ_radon_nikodym_deriv_eq s μ,
+ λ h, h ▸ with_densityᵥ_absolutely_continuous _ _⟩
+
+lemma radon_nikodym_deriv_neg (s : signed_measure α) (μ : measure α) [sigma_finite μ]
+  (hs : s ≪ μ.to_ennreal_vector_measure) :
```

Comment:
Do you really need the assumption `hs` here? Otherwise, there will be a singular part and an absolutely continuous part, and the singular part shouldn't matter for the statement. Same thing for the lemmas below. This makes me wonder if, instead of focusing directly on the absolutely continuous situation, you shouldn't first prove Lebesgue-decomposition-like results in general.

Reply:
I've shown `radon_nikodym_deriv_neg` without the absolutely continuous condition but I'm not sure how to show it for add (not using the same method at least). In particular, I'm not sure if the following is true
```lean
lemma singular_part_add (s t : signed_measure α) (μ : measure α) :
  (s + t).singular_part μ = s.singular_part μ + t.singular_part μ :=
begin
  sorry
end
```
since we can't add Jordan decompositions.

Reply:
Okay, we now have the lebesgue decomposition for signed measures, but to remove the absolutely continuous condition we will also need uniqueness of the singular part. I will try to find a proof for that

Reply:
Done!